### PR TITLE
Improvements from debugging oms system:

### PR DESCRIPTION
**Pull request by palladia**
1. Fixed omiserver memory leak in handling CreateAgentReq
2. Improvements to logfile to distinguish server from engine output
3. Changed missing NTLM message from FATAL to WARNING
4. Modified DumpAllocList() to print address of interest instead of header
5. Removed memory leak of krb5KeytabPath in server
6. Removed trace statements in signal handlers
7. Fixed warning-level log message in Engine's handling of BinNotification msg

### CODE REVIEW

**Comment by palladia** [comment]
@Microsoft/omi-devs

File: Unix/pal/alloc.c [hunk 1]
```diff
@@ -124,7 +124,7 @@ void PAL_DumpAllocList()
 
     for (p = _list; p; p = p->next)
     {
-        trace_DumpAllocList_Block(p->file, (int)p->line, p, p->magic, p->id, p->size);
+        trace_DumpAllocList_Block(p->file, (int)p->line, p+1, p->magic, p->id, p->size);
```
Comment (jeffaco) [inline Unix/pal/alloc.c]: Um, why `p+1`? Minimally, this needs a comment. But it looks like an error, explain please.

File: Unix/protocol/protocol.c [hunk 2]
```diff
@@ -2067,30 +2072,27 @@ static Protocol_CallbackResult _ProcessReceivedMessage(
...
                         handler = newHandler;
 
-                        if(_SendAuthResponse(handler, binMsg->result, binMsg->authFile, forwardSock, 
-                                             binMsg->uid, binMsg->gid))
-                        {
-                            ret = PRT_CONTINUE;
-                        }
+                        if(!_SendAuthResponse(handler, binMsg->result, binMsg->authFile, forwardSock, 
+                                              binMsg->uid, binMsg->gid))
+                            trace_ClientAuthResponseFailed();
```
Comment (jeffaco) [inline Unix/protocol/protocol.c]: Does OMI have a coding standard where all `if` statements should have surrounding braces?

File: Unix/pal/alloc.c (hunk 1 above)
Comment (palladia) [reply Unix/pal/alloc.c]: Added comment to explain +1.

File: Unix/protocol/protocol.c (hunk 2 above)
Comment (palladia) [reply Unix/protocol/protocol.c]: Added braces.

File: Unix/pal/alloc.c [hunk 3]
```diff
@@ -124,7 +124,9 @@ void PAL_DumpAllocList()
 
     for (p = _list; p; p = p->next)
     {
-        trace_DumpAllocList_Block(p->file, (int)p->line, p, p->magic, p->id, p->size);
+        // The address of interest is not the header, hence we print p+1 to
```
Comment (jeffaco) [inline Unix/pal/alloc.c]: s/print/use/:

```
// The address of interest is not the header, hence we use p+1 to
```

**Comment by palladia** [comment]
@paulcallen - please review.

**Pull request merged**